tuna.go

```diff
@@ -298,7 +298,10 @@ func (c *Common) CreateServerConn(force bool) error {
 			if subscribersCount == 0 {
 				return errors.New("there is no service providers for " + c.Service.Name)
 			}
-			offset := rand.Intn(int(subscribersCount))
+			offset := rand.Intn(subscribersCount)
+			if filter != nil && !filter.Empty() {
```

Comment:
You can change the `Empty()` function so it can be called even when filter is nil:
```
function (filter *IPFIlter) Empty() bool {
  if filter == nil {
    return true
  }
  ...
}
```